src/TLuaInterpreter.cpp

```diff
@@ -1985,72 +1985,145 @@ int TLuaInterpreter::getModulePath(lua_State* L)
 {
     QString moduleName;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "getModulePath: Module be be a string");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "getModulePath: bad argument #1 type (module name as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
     } else {
-        moduleName = lua_tostring(L, 1);
+        moduleName = QString::fromUtf8(lua_tostring(L, 1));
     }
+
     Host& host = getHostFromLua(L);
-    QMap<QString, QStringList> modules = host.mInstalledModules;
-    if (modules.contains(moduleName)) {
-        QString modPath = modules[moduleName][0];
-        lua_pushstring(L, modPath.toLatin1().data());
+    if (host.mInstalledModules.contains(moduleName)) {
+        // Windows can use non-ASCII characters in paths easily and POSIX
+        // systems can sometimes be forced to...
+        lua_pushstring(L, host.mInstalledModules.value(moduleName).at(0).toUtf8().constData());
         return 1;
+    } else {
+        lua_pushnil(L);
+        lua_pushfstring(L, "Module named \"%s\" not installed.", moduleName.toUtf8().constData());
+        return 2;
     }
-    return 0;
 }
 
 int TLuaInterpreter::getModulePriority(lua_State* L)
 {
     QString moduleName;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "getModulePriority: Module be be a string");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "getModulePriority: bad argument #1 type (module name as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
     } else {
-        moduleName = lua_tostring(L, 1);
+        moduleName = QString::fromUtf8(lua_tostring(L, 1));
     }
+
     Host& host = getHostFromLua(L);
     if (host.mModulePriorities.contains(moduleName)) {
         int priority = host.mModulePriorities[moduleName];
         lua_pushnumber(L, priority);
         return 1;
     } else {
-        lua_pushstring(L, "getModulePriority: Module doesn't exist");
-        lua_error(L);
+        // Was coded as an hard error (stopping execution of lua command) but
+        // converted to a nil + error message to match other two similar
+        // commands...
+        lua_pushnil(L);
+        lua_pushfstring(L, "Module named \"%s\" not installed.", moduleName.toUtf8().constData());
+        return 2;
+    }
+}
+
+int TLuaInterpreter::getIsModuleSynced(lua_State* L)
```

Comment:
Let's call this just `isModuleSynced()` for consistency with the existing `isActive()`